ksqldb-parser/src/test/java/io/confluent/ksql/parser/SqlFormatterTest.java

```diff
@@ -525,6 +525,34 @@ public void shouldFormatLeftJoinWithWithin() {
     assertEquals(expected, SqlFormatter.formatSql(join));
   }
 
+  @Test
+  public void shouldFormatRightJoinWithoutJoinWindow() {
+    final Join join = new Join(leftAlias, ImmutableList.of(new JoinedSource(
+        Optional.empty(),
+        rightAlias,
+        JoinedSource.Type.RIGHT,
+        criteria,
+        Optional.empty())));
+
+    final String result = SqlFormatter.formatSql(join);
+    final String expected = "`left` L\nRIGHT OUTER JOIN `right` R ON (('left.col0' = 'right.col0'))";
+    assertEquals(expected, result);
+  }
+
+  @Test
+  public void shouldFormatRightJoinWithWithin() {
+    final Join join = new Join(leftAlias, ImmutableList.of(new JoinedSource(
+        Optional.empty(),
+        rightAlias,
+        JoinedSource.Type.RIGHT,
+        criteria,
+        Optional.of(new WithinExpression(10, TimeUnit.SECONDS)))));
+
+    final String expected = "`left` L\nRIGHT OUTER JOIN `right` R WITHIN 10 SECONDS ON "
+        + "(('left.col0' = 'right.col0'))";
+    assertEquals(expected, SqlFormatter.formatSql(join));
+  }
```

Comment:
Side question: do we have any test for `GRACE` clause? We actually want to make `GRACE` mandatory. \cc @spena

Reply:
https://github.com/confluentinc/ksql/pull/8845/commits/319bdd7a345856968b61670dbb2d6a4e877f3040